### PR TITLE
Fix missing set the include directory of gtest in src/codegen/CMakeLists.txt

### PR DESCRIPTION
**Pull request by masayuki038**
When I compiled gandiva/cpp, this error occurred. I don't install gtest package.

```
In file included from /git/gandiva/cpp/src/codegen/expr_decomposer.cc:15:0:
/git/gandiva/cpp/src/codegen/expr_decomposer.h:22:30: fatal error: gtest/gtest_prod.h: No such file or directory
 #include <gtest/gtest_prod.h>
                              ^
```

So added gtest include directory to "target_include_directories" in src/codegen/CMakeLists.txt.

### CODE REVIEW

File: cpp/src/codegen/CMakeLists.txt [hunk 1]
```diff
@@ -55,6 +55,7 @@ target_include_directories(gandiva_obj_lib
     ${CMAKE_SOURCE_DIR}/src
     $<TARGET_PROPERTY:LLVM::LLVM_INTERFACE,INTERFACE_INCLUDE_DIRECTORIES>
     $<TARGET_PROPERTY:ARROW::ARROW_SHARED,INTERFACE_INCLUDE_DIRECTORIES>
+    ${CMAKE_BINARY_DIR}/googletest-src/googletest/include
```
Comment (pravindra) [inline cpp/src/codegen/CMakeLists.txt]: for uniformity, can you please use the same syntax as the dependencies ?

$<TARGET_PROPERTY:gtest_main,INCLUDE_DIRECTORIES>

Comment (masayuki038) [reply cpp/src/codegen/CMakeLists.txt]: Thanks @pravindra for checking.
I have fixed it and also confirmed compilation passed.

**Comment by pravindra** [comment]
@masayuki038  - if you haven't already done this, can you please sign the dremio CLA , and forward a copy to me ?

https://www.dremio.com/legal/contributor-agreement/

**Comment by masayuki038** [comment]
Thanks @pravindra for merging.
I have sent signed document to your mail address.

**Pull request merged**